dylib_shim.cpp

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
```

Comment:
remark: some file like this might be a useful standalone tool, definitely doesn't belong at top-level directory though, will look into moving

Reply:
Potentially a new subdir underneath `/tools/`

Reply:
Yeah--will likely remove from this PR, since this is really dependent on the calling convention between runtime + dylib, which is still WIP in future PR, so I think it makes most sense to merge such a tool there instead